projects/forge-angular/src/lib/dialog/dialog.service.ts

```diff
@@ -18,7 +19,10 @@ export interface IDialogOptions extends Omit<Partial<IDialogComponent>, 'attribu
 @Injectable({
   providedIn: 'root'
 })
-export class DialogService {
+export class DialogService implements OnDestroy {
+  
+  private _openDialogRefs: DialogRef[] = [];
+  private _unsubscribe$ = new Subject();
```

Comment:
Completely optional, since this is functionally equivalent, but since this library requires Angular 16 you could use the [takeUntilDestroyed](https://angular.io/api/core/rxjs-interop/takeUntilDestroyed) operator instead of a custom implementation.

Since it will be called in the `show()` method, which is not an injection context, you will need to inject and pass a reference to the component's `DestroyRef`, but you won't need to implement `ngOnDestroy()`.

Reply:
Were in the process of updating to angular 16.  Thats nice addition.  I added the change.